bvm/ballerina-profiler/src/main/java/io/ballerina/runtime/profiler/ui/JsonParser.java

```diff
@@ -96,12 +90,13 @@ private boolean containsAnySkipList(String str, List<String> arrayList) {
 
     private void cpuParser(ArrayList<String> skipList) {
         try {
-            String file = "CpuPre.json"; // File path of the Profiler Output json file
-            String jsonInput = readFileAsString(file); // Read the json file as a string
+            String jsonInput = FileUtils.readFileAsString("cpu_pre.json"); // Read the json file as a string
 
             // Removes the trailing comma
             StringBuilder jsonInputStringBuffer = new StringBuilder(jsonInput);
-            jsonInputStringBuffer.deleteCharAt(jsonInputStringBuffer.length() - 3);
+            if (jsonInputStringBuffer.length() > 3) {
+                jsonInputStringBuffer.deleteCharAt(jsonInputStringBuffer.length() - 3);
+            }
```

Comment:
Do we need this if condition. If this always true we can remove, Otherwise we need to add the else part also

Reply:
yes. This is to remove the trailing `,` char in the stack trace elements. The `jsonInputStringBuffer.length() > 3` check is used to verify the buffer is not empty.